### DISCUSSION TITLE
Simulating the effect of soil salinity in Date Palm [New model under development]

### DISCUSSION THREAD

**Discussion by KrishnaDevkota** [APSIMInitiative/ApsimX#8835] [General]
Dear Network, 
I am trying to develop new crop "Date Palm" model for APSIM. Date Palm is the major fruit tree in the Mediterranean region. 
Coefficients for Date Palm are customized from Oil Palm model of APSIM. It simulates well the growth and yield of Date Palm, more than 10 coefficients have been parameterized. 
Now, I would like to include the effect of soil salinity in date palms; attached model files. **It did not simulate yield as affected by soil salinity, no yield difference in soil salinity values ranging from 0 to 35 dS/m (every 5 steps). Also, there was no effect of fertilizer on the simulated yield. Can anyone please kindly check the attached model and help ?** 
It's particularly noteworthy that the 'Khalas' variety, under conditions of high water salinity (15 dS/m), produced a markedly low yield of just 3.3 t/ha. The reductions in date yield between treatments with water salinity of 5 and 15 dS/m were observed to be 29%, 43%, and 52% for the varieties ‘Lulu’, ‘Khalas’, and ‘Shahlah’, respectively. Additionally, it's critical to remember that a salinity level of 32 dS/m leads to a complete absence of data palm yield.
I am looking forward to receiving your feedback and support!
[Fert&SalinityEffects.zip](https://github.com/APSIMInitiative/ApsimX/files/14946381/Fert.SalinityEffects.zip)

Best regards, 
Krishna

**Comment by hut104** [comment]
Hi
There is no direct impact of ec on growth in the model.
You will need to account for salinity through parameters such as KL, ll and xf.
I'd suggest working on unaffected trials to parameterise the model first before adding effects of salts.  You may also need to consider how to deal with the transient nature of any salts as well.

**Reply by KrishnaDevkota** [reply]
Dear Neil, 
Thank you very much for your kind suggestion, it is very useful, and highly appreciated.
Best regards, 
Krishna
